### ISSUE TITLE
An option to disable the video history/queue control with the "back" button

### ISSUE THREAD

**Issue by nbmrjuhneibkr** [TeamNewPipe/NewPipe#4479]
### Checklist

- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
v0.20 appears to be keeping the history/queue of previously opened videos separately from the actual browsing history.
As discussed in https://github.com/TeamNewPipe/NewPipe/issues/4462 , this interferes with the normal behavior of the Back button, and may be inconvenient and counterintuitive.  
The solution would be to add an option to always treat the Back button the same way as it's treated in any web browsing application, and don't control the video queue with it.

#### Is your feature request related to a problem? Please describe it
Unless the minimized player is closed manually (which also stops any playback), Back button will always open previous video (even if autoplay is disabled, and user never actually watched the video) instead of going to the previous page (playlist/channel/video/search results).

#### Additional context
https://github.com/TeamNewPipe/NewPipe/issues/4462#issuecomment-706205009

#### How will you/everyone benefit from this feature?
This will allow to keep traditional navigation with Back button always leading to the previously opened page.

**Comment by avently** [comment]
Finally. You made a normally described issue

**Comment by nbmrjuhneibkr** [comment]
>Finally. You made a normally described issue

Am I getting a medal for that or something? All issues submitted by me are made in compliance with the contribution guidelines.

**Comment by FriendlyNeighborhoodShane** [comment]
Besides changing the behaviour of the back button, IMO the video should *not* automatically be added to this queue at all unless it has actually been played (by autoplay or otherwise).

**Comment by maxily** [comment]
My two cents. One of the confusions i have with the queue is that videos i watched haven't been removed from the queue automatically after they completed, so when i press back (device's back button) i usually navigated through the stack of finished videos, which is slightly annoying. I'd prefer that navigation from current video to be leading to Channel and then to Subscriptions list, without implicit queue.

If the queue is still desired, then it would be great if the interaction with it was more transparent. From what it sounds like, seems it should be a separate feature with explicit submenus like "add to/remove from queue" with a separate tab for the list maybe.
For example, how it's done in AntennaPod, see a [screenshot ](https://raw.githubusercontent.com/AntennaPod/AntennaPod/develop/app/src/main/play/listings/en-US/graphics/phone-screenshots/03.png) - the app has a Queue list which basically contains current playlist of episodes to play, where user can edit the order etc

**Comment by MD77MD** [comment]
I would like to have the back button as well but not in current newPipe status.  the video queue history is just essential in the current newPipe structure and this why. newPipe lakes the multiple windows/fragments the orgenal YouTube app have, as you know we could have up to 5 search instances that we can interact and switch between them. while newPipe can handle only one search instant. in my experience (which is actually a work around) is the queue history helps to mitigate the lake of this hey feature by holding videos i have searched and opened back to back then go through them one by one. 

I know this is a stupid work around but what to do, however, if the multi-window feature is added (in the next +5 years, maybe) then i think we can do something about it. perhaps a toggle in options would do. but to be honest i would still love  to have it. 


@maxily your idea for have a queue panel is even more important. to be honest I've been having similarly thoughts. a separate independent queue panel that could be accessed easily fun anywhere in the app, not from the notification panel. moreover, we also most have "add next" for this to function properly. Maybe you can open an new issue for this

**Comment by gregor-tb** [comment]
I also struggle with the back / history.

Open a video from feed, "close" it with back button minimizes the player, what is a nice feature, you can hear the outro and browse the next video. 

BUT when you don't close the Mini-Player by yourself + open a new video from feed, you expect to go back to feed (with minimized player etc.). But instead you get back to the previous video detail page, that is very annoying. No matter if you watched the video or just visited the page, starting from feed should not enqueue (or be optional).

An overall present "back to feed" button would also fix the issue, because going back in history through dozen search pages is also frustrating, I restart the app then, it's faster... :(

**Comment by avently** [comment]
@gregor-tb 
>An overall present "back to feed" button would also fix the issue, because going back in history through dozen search pages is also frustrating, I restart the app then, it's faster... :(

Yep, it's much faster than to close the mini player with gesture down or via tap on X button. Nice idea!

**Comment by eudes** [comment]
I understand the technical difficulties in making the back button work as requested, but the current behaviour is totally contrary to mobile app UX idioms and should be reverted. No one expects the back button to take you to something different than the pervious screen. This only happens with obtrusive ad focused websites, and it annoys the hell out of everyone.

The expected behaviour of the back button is to take you back to the last screen, whatever that is. The "the take me to the previous video" verb is not something anyone expects from the native buttons, and should be made explicit by using something like a "previous track" button on the video overlay.

If decoupling this UI feature from the queue mechanism is impossible, then I suspect the queue mechanism is not particularly well designed and should be given some more thought.

**Comment by Vrihub** [comment]
> I understand the technical difficulties in making the back button work as requested, but the current behaviour is totally contrary to mobile app UX idioms and should be reverted

I agree in principle, but if reverting would cause too much disruption, or require too much refactoring of the existing code, then until a solution is found, I think this non-standard behaviour ("swipe down" is the new "back") should at least be well documented, so that the user can be familiar with it from day one (instead of being puzzled and learn it with frustration).

By "documented" I mean with a help screen that would appear the first time the app is run (or the first time the back button is used :)); perhaps also a "question mark" icon somewhere in the main video screen could lead to this help screen, like the one we currently have in the "What's new" screen to explain "fast mode".

**Comment by nbmrjuhneibkr** [comment]
Any updates on this? @eudes said it best: current behavior of the _system button_ in NewPipe contradicts basic mobile UX. At the very least, this non-standard behavior should be optional and/or be handled by a custom UI element - not by a system button.

**Comment by SameenAhnaf** [comment]
Hack at this moment. Swipe down the upper interface instead of clicking back button.

https://user-images.githubusercontent.com/79245641/117391050-0a79ee80-af11-11eb-9cf9-bd823ecd319f.mp4

**Comment by opusforlife2** [comment]
> Hack at this moment. Swipe down the upper interface instead of clicking back button.

@SameenAhnaf That's not a hack. That is the actual intended navigation flow. xD

Have you not read the 0.20.0 blog post?

**Comment by SameenAhnaf** [comment]
@opusforlife2 That's true. What I intended to say is that back button is unusable at this moment. Typically, we use 'Back' button again and again which takes us into the problem. However, this hassle could be handled at least a little.

You can either use 'Back' button while exiting next video continuously and get disturbed by previous video infoes again and again. Or, use 'Swipe down gesture' while minimizing the video and get back into video player.

Browsing on portrait mode is nearly impossible at this moment. Probably, the word 'Hack' wasn't a proper way to express it.

**Comment by SameenAhnaf** [comment]
@opusforlife2 That's true. What I intended to say is that back button is unusable at this moment. Typically, we use 'Back' button again and again which takes us into the problem. However, this hassle could be handled at least a little.

You can either use 'Back' button while exiting next video continuously and get disturbed by previous video infoes again and again. Or, use 'Swipe down gesture' while minimizing the video and get back into video player.

Browsing on portrait mode is nearly impossible at this moment. Probably, the word 'Hack' wasn't a proper way to express it.

>I think this non-standard behaviour ("swipe down" is the new "back") should at least be well documented, so that the user can be familiar with it from day one (instead of being puzzled and learn it with frustration).

Lol, I didn't notice this! I should not have skimmed the texts.

**Comment by avently** [comment]
@SameenAhnaf do you know that you can edit your messages instead of deleting and recreating them? Or it's a hack for you too? Why do we need to receive dozens of emails because if you want to receive an attention?

**Comment by SameenAhnaf** [comment]
@avently Extremely sorry for that. But I still receive mails and notificationseven if the comments are edited. I am really not sure what is wrong with my GitHub account. I noticed that a couple of times.

Do you know any way to stop that?

**Comment by avently** [comment]
@SameenAhnaf I don't receive an email if someone edits their comment so can't say what you can do. Maybe you can try to write to github support they are pretty fast

**Comment by ShrimpBFM** [comment]
I barely use the app anymore since this nasty mini player was introduced because of the issue described here and also the fact 
that I don't want the mini player all.

There should be an option to disable it. 
When I watch a video and press the back button, the video should close and I should get back to my search results or subscription feed (where ever I came from.)

I neither do I want some video sound going on distract me while I browse for the next video, nor do I want to tap a little X on my phone with my thumb unsuccessfully.

Not having such annoyances used to be the main reason to use this app over regular YouTube to me.
But then you added extra steps and annoyances into the experience with the introduction of this mandatory "feature".

I check the app every couple of months and it is still not optional.

Please make the app user-friendly again.

**Comment by opusforlife2** [comment]
@ShrimpBFM You can swipe down on the video with two fingers to close it completely.

**Comment by Multipanda** [comment]
This has now become worse, rather than having to click back a few times, if I've watched a bunch of videos through the suggestion feed i now have to click back through **every single one** making the app bordeline useless if I want to get back to the app's start screen.

**Comment by nbmrjuhneibkr** [comment]
@opusforlife2 This is non-intuitive, and can't be done with one hand. Not a viable solution for the problem.

**Comment by avently** [comment]
Made a PR with a fix, take a look: #6895

**Comment by eudes** [comment]
Gave it a quick test and it's all I was hoping for :). Thank you!

For anyone wanting to try it: download the build artifact, which contains the APK; install; change the following setting (in the app) to you preferred action: Settings > Video and Audio > Behaviour > Back press in player. The YouTube app-like behaviour is called "Hide into the mini player".

This is the artifact with this feature (as of right now): https://github.com/TeamNewPipe/NewPipe/suites/3461129393/artifacts/82255089
You may need to be logged in to github to download it.

**Comment by Schnyarf** [comment]
I, too think that this is important for concerns related to accessibility, consistency with other video browsers such as mainline YT, and consistency with the rest of the Android UX.

With the exception of its implementation in things like web browsers, the 'back' gesture isn't really meant to navigate backward through content that is being browsed—we would then expect there to be a 'forth' button as well. While it can be inconsistent at times, it is principally meant to take the user back to the previous UI scene.

This is why the YT app behavior of minimizing the video player is better in my opinion, because it's more consistent with that standard. Back-and-forth media history navigation should be a thing reserved for media-control buttons, and it should be *uncommon to trigger accidentally*.

And just from an accessibility and ease-of-use standpoint, I think minimizing the player to go search is a principal action that shouldn't have to involve reaching up and gesture-controlling all the time—phones are getting huge these days :P.

**Comment by Jon-guy30** [comment]
I hope this is the right GitHub issue for this.

Was this ever fixed, why is this behaviour still present? It's still here on version 0.26.1. I even tried out the unofficial sponsorblock version and it's there too, I would have hoped the OTHER author would have fixed it, but nope, they just copied NewPipe's code and behaviour. 

Sigh...

I keep trying out newpipe but I'm still stuck watching YouTube on Firefox with uBlock Origin because I hate this non-standard mobile UX behaviour.

When I do the back gesture in Android 12 I expect the video to close, not minimize into this list kind of thing, I guess it's the mini-player or whatever it is. I have to do multiple gestures, first the standard back gesture navigation and then hitting the little x to close the video.

I was under the impression that it was fixed or at least a fix was on its way, am I missing something? It's not in the settings, I looked.

**Comment by burgrr** [comment]
> I hope this is the right GitHub issue for this.
> 
> Was this ever fixed, why is this behaviour still present? It's still here on version 0.26.1. I even tried out the unofficial sponsorblock version and it's there too, I would have hoped the OTHER author would have fixed it, but nope, they just copied NewPipe's code and behaviour.
> 
> Sigh...
> 
> I keep trying out newpipe but I'm still stuck watching YouTube on Firefox with uBlock Origin because I hate this non-standard mobile UX behaviour.
> 
> When I do the back gesture in Android 12 I expect the video to close, not minimize into this list kind of thing, I guess it's the mini-player or whatever it is. I have to do multiple gestures, first the standard back gesture navigation and then hitting the little x to close the video.
> 
> I was under the impression that it was fixed or at least a fix was on its way, am I missing something? It's not in the settings, I looked.

try using Revanced, it's the only one with devs that can take less than five years to make a slight UI change
